### ISSUE TITLE
Field components have inconsistent spacing between label and field component

### ISSUE THREAD

**Issue by AmandaFurukawa** [carbon-design-system/carbon-design-kit#699]
## Detailed description

In the figma library - carbon V11

There is inconsistent spacing set between the label and field components (dropdown, input, number input, text area input).
It looks like some of the labels have extra padding set which is causing the spacing to be more than it should be. 

> What version of the Carbon Design System Kit are you using?
V11

> Is this issue related to a specific component? If so please give the page, artboard, and layer/symbol name.
default - dropdown, text input, area input, number input

> What did you expect to happen? What happened instead? What would you like to see changed?
They should all have the same spacing between the label and the field component

> Is this issue being filed based on a discrepancy between the website/code and the kit? If so what is the inconsistency?

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

## Steps to reproduce the issue

1. pull all 4 components in figma from v11 library and place them all beside each other to see inconsistent spacing

## Additional information

<img width="1006" alt="image" src="https://github.com/carbon-design-system/carbon-design-kit/assets/119629745/040e0470-0d9e-4338-8849-5cbc4faffabd">

the label margin of number input, dropdown have below padding set to 4 (text input is set to no extra padding)
<img width="1692" alt="image" src="https://github.com/carbon-design-system/carbon-design-kit/assets/119629745/6c0e114c-8443-4ea1-ae23-11f4a8dc8c19">

text area has 8 spacing for the label/text area (which i guess is overriding the text area overall 4px spacing) 
<img width="1712" alt="image" src="https://github.com/carbon-design-system/carbon-design-kit/assets/119629745/6631837a-c4ad-4dab-87de-293afd08e774">

## Add labels
Please choose the appropriate label(s) from our existing label list to ensure that your issue is properly categorized (i.e.use the `libraries` label for any issue relating to the Carbon Sketch Library). This will help us to better understand and address your issue.

**Comment by AmandaFurukawa** [comment]
![image](https://github.com/carbon-design-system/carbon-design-kit/assets/119629745/2923b07c-0d17-4f87-b3f1-e7715f514fbc)
I see that 8px is the proper spacing between the label and field so text input is the one that is incorrect (its 4) - but also the way the spacing is set up for all components are not consistent (FYI)

**Comment by aagonzales** [comment]
Here's how the spacing should work in Figma, the 4px problem was coming from the helper text spec. I also see the spacing set-up inconsistency that she mentioned.  

<img width="666" alt="image" src="https://github.com/carbon-design-system/carbon-design-kit/assets/11670886/9c8c7d72-c598-4dfd-a9ab-26b623071feb">

<img width="659" alt="image" src="https://github.com/carbon-design-system/carbon-design-kit/assets/11670886/4477da8f-e551-4653-9660-5e45a6384f3b">

**Comment by aagonzales** [comment]
Ok I have updated the `text input` labels to have the total 8px bottom padding and the fix will be available in the next library publish. 

However, concerning the other problem with inconsistent spacing here is what I found out. There are two ways of doing label spacing

1. 4px vertical spacing + 4px padding bottom
Dropdowns, Number input, Select, Password input

<img width="696" alt="image" src="https://github.com/carbon-design-system/carbon-design-kit/assets/11670886/7e228bf1-ccad-46f0-9463-93c04fae1f5f">


2. 8px padding-bottom on the label
Text input, text area, date picker

<img width="757" alt="image" src="https://github.com/carbon-design-system/carbon-design-kit/assets/11670886/f1be403a-7cc4-43fd-9a6f-d7c6fb221097">
